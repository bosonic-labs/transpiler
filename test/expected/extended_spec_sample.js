(function () {
    window.SuperTickTockClock = document.registerElement('super-tick-tock-clock', {
        prototype: Object.create(TickTockClock.prototype, {
            tick: {
                enumerable: true,
                value: function () {
                    TickTockClock.prototype.tick.call(this);
                    var now = new Date();
                    this.style.borderColor = now.getSeconds() % 2 ? '#ccc' : 'red';
                }
            }
        })
    });
}());