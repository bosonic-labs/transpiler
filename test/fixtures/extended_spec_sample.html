<element name="super-tick-tock-clock" extends="tick-tock-clock">
  <script>
    ({
      tick: function () {
        super.tick();
        var now = new Date();
        this.style.borderColor = now.getSeconds() % 2 ? '#ccc' : 'red';
      }
    });
  </script>
</element>