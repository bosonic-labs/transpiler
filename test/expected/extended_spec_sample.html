<element name="super-tick-tock-clock" extends="tick-tock-clock">
  <script>
(function () {
    window.SuperTickTockClock = document.registerElement('super-tick-tock-clock', {
        prototype: Object.create(TickTockClock.prototype, {
            tick: {
                enumerable: true,
                value: function () {
                    this._super.tick.call(this);
                    console.log('tick');
                }
            }
        }),
        extends: 'tick-tock-clock'
    });
    Object.defineProperty(SuperTickTockClock.prototype, '_super', {
        enumerable: false,
        writable: false,
        configurable: false,
        value: TickTockClock.prototype
    });
}());
</script>
</element>